### PR TITLE
Wait for blob inclusion

### PR DESCRIPTION
**Pull request by ak88**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [x] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

Send blobs tool now includes: 

- Command line parsing of args.
- Ability to distribute funds to many private keys and use a different sender for each blob tx.
- Reclaiming the distributed funds when testing is done. 
- Ability to wait for block inclusion before posting the next. 
- Args for settings gas parameters explicitly.
- Moved project to the tools directory.
- Own solution file.

### CODE REVIEW

**Review by flcl42 — CHANGES_REQUESTED** [review]

File: tools/Nethermind.SendBlobs/Program.cs [hunk 1]
```diff
@@ -184,10 +202,40 @@
...
+                Console.WriteLine($"Found blob transaction in block {blockResult.Number}");
+                return;
+            }
+        }
+        else
+        {
+            await Task.Delay(2000);
+        }
     }
 }
```
Comment (flcl42) [inline tools/Nethermind.SendBlobs/Program.cs:241]: Could also run `dotnet format whitespace . --folder` in the project folder and change `var` to types?

File: tools/Nethermind.SendBlobs/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy AS build
 COPY . .
-RUN dotnet publish ./src/Nethermind/Nethermind.SendBlobs/Nethermind.SendBlobs.csproj --sc -o out
+RUN dotnet publish ./tools/Nethermind.SendBlobs/Nethermind.SendBlobs.csproj --sc -o out
```
[the remark this answers is not in the document]
Comment (ak88) [reply tools/Nethermind.SendBlobs/Dockerfile]: Renamed to just SendBlobs now

File: tools/Nethermind.SendBlobs/Program.cs [hunk 3]
```diff
@@ -184,10 +202,40 @@
         string txRlp = Hex.ToHexString(txDecoder
             .Encode(tx, RlpBehaviors.InMempoolForm | RlpBehaviors.SkipTypedWrapping).Bytes);
 
+        BlockModel<Keccak>? blockResult = null;
+        if (waitForBlobInclusion)
+            blockResult = await nodeManager.Post<BlockModel<Keccak>>("eth_getBlockByNumber", "latest", false);
+       
         string? result = await nodeManager.Post<string>("eth_sendRawTransaction", "0x" + txRlp);
```
[the remark this answers is not in the document]
Comment (ak88) [reply tools/Nethermind.SendBlobs/Program.cs]: Added a retry count for the wait loop

File: tools/Nethermind.SendBlobs/Program.cs (hunk 1 above)
Comment (ak88) [reply tools/Nethermind.SendBlobs/Program.cs]: Changed as suggested

File: src/Nethermind/Nethermind.sln [hunk 4]
```diff
@@ -215,7 +215,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Nethermind.Serialization.Ss
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Nethermind.Network.Contract", "Nethermind.Network.Contract\Nethermind.Network.Contract.csproj", "{AD151E35-4BBC-4A83-8F57-CC8665F6E007}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Nethermind.SendBlobs", "Nethermind.SendBlobs\Nethermind.SendBlobs.csproj", "{E2CEB1D9-A81C-4142-9324-E086FF312636}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SendBlobs", "..\..\tools\SendBlobs\SendBlobs.csproj", "{EAA543AC-CD70-4C8E-925C-3E2D878ECA89}"
```
Comment (rubo) [inline src/Nethermind/Nethermind.sln:218]: Linking a project to the main solution that's not in the `src` directory is not a good approach. Either it moves to `src` or gets excluded from the solution. The latter may be.

**Review by rubo — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.sln (hunk 4 above)
Comment (ak88) [reply src/Nethermind/Nethermind.sln]: Removed the project from main solution

File: src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs [hunk 5]
```diff
@@ -314,7 +314,7 @@ private void EncodeShardBlobPayloadWithoutPayload(T item, RlpStream stream, RlpB
             stream.Encode(item.Value);
             stream.Encode(item.Data);
             _accessListDecoder.Encode(stream, item.AccessList, rlpBehaviors);
-            if(item.MaxFeePerDataGas.Value == 42_000_000_000)
+            if (item.MaxFeePerDataGas.Value == 42_000_000_000)
```
Comment (flcl42) [inline src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs:317]: Oh I've almost forgot that we break the client, let's think how we can do the same without adding bugs

**Review by MarekM25 — CHANGES_REQUESTED** [review]
We can't merge with this line in TxDecoder         if (item.MaxFeePerDataGas.Value == 42_000_000_000)

File: src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs (hunk 5 above)
Comment (ak88) [reply src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs]: Rolled back

**Review by flcl42 — APPROVED** [review]

**Pull request closed without merging**